File: again/pascalsTriangle.js
const pascalsTriangle = numRows => {
    if (!numRows) {
        return [];
    }

    if (numRows === 1) {
        return [[1]];
    }

    if (numRows === 2) {
        return [[1], [1, 1]];
    }

    const triangle = [[1], [1, 1]];

    for (let i = 2; i < numRows; i++) {
        let j = 0;
        const iter = i - 1;
        const a = [1];

        while (j < iter) {
            a.push(triangle[i - 1][j] + triangle[i - 1][j + 1]);
            j++;
        }

        a.push(1);

        triangle.push(a);
    }

    return triangle;
};

console.log(
    pascalsTriangle(10)
);

File: python/recursion/pascals_triangle.py
# Iterative (dynamic programming b/c not adding the outside 1s every time)
def pascals_triangle(n):
    triangle = []

    for i in range(n):
        row = [None] * (i + 1)
        row[0], row[-1] = 1, 1

        for j in range(1, len(row) - 1):
            row[j] = triangle[i - 1][j - 1] + triangle[i - 1][j]

        triangle.append(row)

    return triangle

print(pascals_triangle(7))
